libs/model/src/policies/DiscourseImportWorker.policy.ts

```diff
@@ -0,0 +1,364 @@
+import {
+  createAllAddressesInCW,
+  createAllCommentsInCW,
+  createAllReactionsInCW,
+  createAllThreadsInCW,
+  createAllTopicsInCW,
+  createAllUsersInCW,
+  importDump,
+} from '../services';
+import { CleanupFn, runCleanup } from '../utils';
+
+const log = logger(import.meta);
+
+const Errors = {
+  CommunityNotFound: 'community not found',
+};
+
+const inputs = {
+  DiscourseImportSubmitted: events.DiscourseImportSubmitted,
+};
+
+export function DiscourseImportWorker(): Policy<typeof inputs> {
+  return {
+    inputs,
+    body: {
+      DiscourseImportSubmitted: async ({ payload }) => {
+        const { id: communityId } = payload;
+
+        const community = await models.Community.findByPk(communityId);
+        if (!community) {
+          throw new AppError(Errors.CommunityNotFound);
+        }
+        const { base, dumpUrl } = payload;
+
+        // cleanup functions are pushed to this array, then popped off
+        // and invoked after everything is done
+        const cleanupStack: CleanupFn[] = [];
+
+        let restrictedDiscourseConnection: Sequelize | null = null;
+
+        try {
+          const now = Date.now();
+
+          const discourseDbName = `temp_discourse_dump_${now}`;
+          const restrictedDiscourseDbUser = `temp_discourse_importer_${now}`;
+          const restrictedDiscourseDbPass = `temp_discourse_importer_pass_${now}`;
+
+          // create discourse DB user
+          await models.sequelize.query(
+            `CREATE ROLE ${restrictedDiscourseDbUser} WITH LOGIN PASSWORD '${restrictedDiscourseDbPass}';`,
+          );
+          cleanupStack.push({
+            description: 'Drop discourse DB user',
+            fn: async () => {
+              await models.sequelize.query(
+                `DROP ROLE ${restrictedDiscourseDbUser};`,
+              );
+            },
+          });
+
+          // create discourse DB
+          await models.sequelize.query(`CREATE DATABASE ${discourseDbName};`);
+          cleanupStack.push({
+            description: 'Drop discourse DB',
+            fn: async () => {
+              await models.sequelize.query(`
+            SELECT pg_terminate_backend(pid)
+            FROM pg_stat_activity
+            WHERE datname = '${discourseDbName}' AND pid <> pg_backend_pid();
+          `);
+              await models.sequelize.query(`DROP DATABASE ${discourseDbName};`);
+            },
+          });
+
+          // connect to discourse DB as superuser to
+          // grant privileges to restricted discourse user
+          const superUserDiscourseDbUri = (() => {
+            const parsedUrl = new URL(config.DB.URI);
+            parsedUrl.pathname = discourseDbName;
+            if (config.DB.NO_SSL) {
+              parsedUrl.searchParams.set('sslmode', 'disable');
+            }
+            return parsedUrl.toString();
+          })();
+          const superUserDiscourseConnection =
+            await createDiscourseDBConnection(superUserDiscourseDbUri);
```

Comment:
Correct me if I am wrong here, but with this current implementation we would be creating a new sequelize client with a new connection pool every invocation of this policy? Maybe it is better to use the global variable approach like we do with our normal sequelize client.